main.go

```diff
@@ -0,0 +1,205 @@
+	os.Exit(1)
+}
+
+func main() {
+	var cfg config
+	if err := stepconf.Parse(&cfg); err != nil {
+		failf("Error: %s\n", err)
+	}
+	stepconf.Print(cfg)
+
+	// Change dir to working directory
+	workDir, err := pathutil.AbsPath(cfg.WorkDir)
+	if err != nil {
+		failf("Failed to expand WorkDir (%s), error: %s", cfg.WorkDir, err)
+	}
+
+	currentDir, err := pathutil.CurrentWorkingDirectoryAbsolutePath()
+	if err != nil {
+		failf("Failed to get current directory, error: %s", err)
+	}
+
+	if workDir != currentDir {
+		fmt.Println()
+		log.Infof("Switch working directory to: %s", workDir)
+
+		revokeFunc, err := pathutil.RevokableChangeDir(workDir)
+		if err != nil {
+			failf("Failed to change working directory, error: %s", err)
+		}
+		defer func() {
+			fmt.Println()
+			log.Infof("Reset working directory")
+			if err := revokeFunc(); err != nil {
+				failf("Failed to reset working directory, error: %s", err)
+			}
+		}()
+	}
+
+	// Update cordova and ionic version
+	if cfg.CordovaVersion != "" {
+		fmt.Println()
+		log.Infof("Updating cordova version to: %s", cfg.CordovaVersion)
+
+		if err := npm.Remove(false, "cordova"); err != nil {
+			failf("Failed to remove cordova, error: %s", err)
+		}
+
+		if err := npm.Install(true, "cordova@"+cfg.CordovaVersion); err != nil {
+			failf("Failed to install cordova, error: %s", err)
+		}
+	}
+
+	if cfg.IonicVersion != "" {
+		fmt.Println()
+		log.Infof("Updating ionic version to: %s", cfg.IonicVersion)
+
+		if err := npm.Remove(false, "ionic"); err != nil {
+			failf("Failed to remove ionic, error: %s", err)
+		}
+
+		if err := npm.Install(true, "ionic@"+cfg.IonicVersion); err != nil {
+			failf("Failed to install ionic, error: %s", err)
+		}
+
+		fmt.Println()
+		log.Infof("Installing local ionic cli")
+		if err := npm.Install(false, "ionic@"+cfg.IonicVersion); err != nil {
+			failf("command failed, error: %s", err)
+		}
+	}
+
+	// Print cordova and ionic version
+	cordovaVer, err := cordova.Version()
+	if err != nil {
+		failf("Failed to get cordova version, error: %s", err)
+	}
+
+	fmt.Println()
+	log.Printf("cordova version: %s", colorstring.Green(cordovaVer.String()))
+
+	ionicVer, err := ionic.Version()
+	if err != nil {
+		failf("Failed to get ionic version, error: %s", err)
+	}
+
+	log.Printf("ionic version: %s", colorstring.Green(ionicVer.String()))
+
+	// Ionic CLI plugins angular and cordova have been marked as deprecated for
+	// version 3.8.0 and above.
+	ionicVerConstraint, err := ver.NewConstraint("< 3.8.0")
+	if err != nil {
+		failf("Could not create version constraint for ionic: %s", err)
+	}
+	if ionicVerConstraint.Check(ionicVer) {
+		fmt.Println()
+		log.Infof("Installing cordova and angular plugins")
+		if err := npm.Install(false, "@ionic/cli-plugin-ionic-angular@latest", "@ionic/cli-plugin-cordova@latest"); err != nil {
+			failf("command failed, error: %s", err)
+		}
+	}
+
+	ionicMajorVersion := ionicVer.Segments()[0]
+
+	platforms := strings.Split(cfg.Platform, ",")
```

Comment:
this must be enough if you have stepconf validation and also a fixed value_options item. also what strings.sort for in this case?